### PR TITLE
[Fix] Backing up EEPROM before performing OTA For fixing - [#1808]

### PR DESCRIPTION
**Pull request by arihantdaga**
This is for taking backup of eeprom into last sector (base sector) before updating the firmware. Is helpful for builds with no SPIFFS.

### CODE REVIEW

File: code/espurna/nofuss.ino [hunk 1]
```diff
@@ -145,6 +145,7 @@ void nofussSetup() {
             #if WEB_SUPPORT
                 wsSend_P(PSTR("{\"action\": \"reload\"}"));
             #endif
+            eepromBackup(0);
```
Comment (mcspr) [inline code/espurna/nofuss.ino]: My original proposal was to insert this into the eepromRotate false condition

**Comment by arihantdaga** [comment]
Yeah and i think i did another mistake also, this isnt the right place to keep eepromBackup, it should be kept in NOFUSS_UPDATING hook. 
I'll do another commit and fix everything.

**Comment by mcspr** [comment]
Just to reiterate, it should do forced backup with nofuss (eepromBackup(0)) right after the rotation is disabled and there should be eepromCommit(false) in the eepromRotate (since it marks eeprom as _dirty, it will force a write next loop()).

You are correct that it is better to write before any flash writes from Updater (while EEPROM buffer is in memory, just to protect from any issues during the update or other accidental failures)

**Comment by mcspr** [comment]
I think I covered all of the options? I have tried some builds without SPIFFS by using ArduinoOTA upgrade method (it does not leave loop(), same as nofuss) and upgrade triggered the rotation backup.
There a small unrelated fixes pending too, so I wanted to merge this right now. Thanks for pointing out the issue!

**Pull request merged**